cpp/cmake/thirdparty/get_libcugraphops.cmake

```diff
@@ -16,20 +16,63 @@
 
 set(CUGRAPH_MIN_VERSION_cugraph_ops "${CUGRAPH_VERSION_MAJOR}.${CUGRAPH_VERSION_MINOR}.00")
 
+if(NOT DEFINED CUGRAPH_CUGRAPH_OPS_VERSION)
+  set(CUGRAPH_CUGRAPH_OPS_VERSION "${CUGRAPH_VERSION_MAJOR}.${CUGRAPH_VERSION_MINOR}")
+endif()
+
+if(NOT DEFINED CUGRAPH_CUGRAPH_OPS_BRANCH)
+  set(CUGRAPH_CUGRAPH_OPS_BRANCH "branch-${CUGRAPH_CUGRAPH_OPS_VERSION}")
+endif()
+
+if(NOT DEFINED CUGRAPH_CUGRAPH_OPS_REPOSITORY)
+  set(CUGRAPH_CUGRAPH_OPS_REPOSITORY "git@github.com:rapidsai/cugraph-ops.git")
+endif()
+
 function(find_and_configure_cugraph_ops)
 
-    set(oneValueArgs VERSION)
+    set(oneValueArgs VERSION REPO PINNED_TAG BUILD_STATIC EXCLUDE_FROM_ALL CLONE_ON_PIN)
     cmake_parse_arguments(PKG "" "${oneValueArgs}" "" ${ARGN})
 
-    rapids_find_package(cugraph-ops ${PKG_VERSION} REQUIRED
+    if(PKG_CLONE_ON_PIN AND NOT PKG_PINNED_TAG STREQUAL "branch-${CUGRAPH_CUGRAPH_OPS_VERSION}")
+        message("Pinned tag found: ${PKG_PINNED_TAG}. Cloning cugraph-ops locally.")
+        set(CPM_DOWNLOAD_cugraph-ops ON)
+    elseif(PKG_BUILD_STATIC AND (NOT CPM_cugraph-ops_SOURCE))
+      message(STATUS "CUGRAPH: Cloning cugraph-ops locally to build static libraries.")
+      set(CPM_DOWNLOAD_cugraph-ops ON)
+    endif()
+
+    set(CUGRAPH_OPS_BUILD_SHARED_LIBS ON)
+    if(PKG_BUILD_STATIC)
+      set(CUGRAPH_OPS_BUILD_SHARED_LIBS OFF)
+    endif()
+
+    rapids_cpm_find(cugraph-ops ${PKG_VERSION} REQUIRED
       GLOBAL_TARGETS      cugraph-ops::cugraph-ops++
       BUILD_EXPORT_SET    cugraph-exports
       INSTALL_EXPORT_SET  cugraph-exports
+      CPM_ARGS
+        SOURCE_SUBDIR    cpp
+        GIT_REPOSITORY   ${PKG_REPO}
+        GIT_TAG          ${PKG_PINNED_TAG}
+        EXCLUDE_FROM_ALL ${PKG_EXCLUDE_FROM_ALL}
+        OPTIONS
+          "BUILD_CUGRAPH_OPS_CPP_TESTS OFF"
+          "BUILD_SHARED_LIBS ${CUGRAPH_OPS_BUILD_SHARED_LIBS}"
     )
 endfunction()
 
 ###
+# Change pinned tag and fork here to test a commit in CI
+#
 # To use a locally-built cugraph-ops package, set the CMake variable
 # `-D cugraph-ops_ROOT=/path/to/cugraph-ops/build`
 ###
-find_and_configure_cugraph_ops(VERSION ${CUGRAPH_MIN_VERSION_cugraph_ops})
+find_and_configure_cugraph_ops(VERSION      ${CUGRAPH_MIN_VERSION_cugraph_ops}
+                               REPO         ${CUGRAPH_CUGRAPH_OPS_REPOSITORY}
+                               PINNED_TAG   ${CUGRAPH_CUGRAPH_OPS_BRANCH}
+                               BUILD_STATIC     ${CUGRAPH_USE_CUGRAPH_OPS_STATIC}
+                               EXCLUDE_FROM_ALL ${CUGRAPH_EXCLUDE_CUGRAPH_OPS_FROM_ALL}
+                               # When PINNED_TAG above doesn't match cugraph,
+                               # force local cugraph-ops clone in build directory
+                               # even if it's already installed.
```

Comment:
How does this affect non-RAPIDS devs that may not have access to the cugraph-ops (closed) source?  Rather than attempting a clone by default, should it output an informative error instead?  I haven't tried this, but I'm concerned that this would output a message about invalid credentials and the developer wouldn't know why or what happened.

Reply:
That is true. Previously we were just doing `rapids_find_package` here so there was no risk of that. One option would be to add a separation option like `ALLOW_CLONE_CUGRAPH_OPS` to control whether or not we use a `rapids_find_package` call instead of `rapids_cpm_find`. Do you think that's worth doing (and if so, do you think it's worth doing in this PR, or saved for a follow-up)?

Reply:
I think an `ALLOW_CLONE_CUGRAPH_OPS` which defaults to `OFF` would be good to prevent unexpected clone attempts of the closed source repo, and possibly go a long way in preventing issues filed from community members building from source.  It's up to you if you think it's better in this PR or a separate one, since I image we want this merged soon.  I can file an issue for this if it's going in a separate PR.

Reply:
https://github.com/rapidsai/cugraph/issues/3004

Reply:
Since I had to fix merge conflicts and wait for one more round of CI I went ahead and implemented this flag in this PR.